helpers/mod_api_helper.rb

```
@@ -78,8 +78,42 @@ def find_artefact(artefact_id)
+            Array(doc["ontology_viewingRestriction_txt"]).any? {|u| u.split(' ').last == current_user&.username} ||
+            Array(doc["ontology_acl_txt"]).any? {|u| u.split(' ').last == current_user&.username}
+
+          user_restricted = !user_not_restricted
+          next if user_restricted
+
+          acronyms_ids[acronym] = id
+          result[acronym] = LinkedData::Models::SemanticArtefact.read_only(id: "#{LinkedData.settings.id_url_prefix}artefacts/#{acronym}", acronym: acronym, description: doc['description_text'], title: doc['ontology_name_text'])
+        end
+
+        return hydra_page_object(result.values, result.length)
+      end
```

RuboCop (1.75.5)

[convention] Assignment Branch Condition size for search_metadata is too high. [<19, 44, 22> 52.74/23]

(Metrics/AbcSize)